oxd-gen-client/src/test/java/io/swagger/client/api/GetTokensByCodeTest.java

```diff
@@ -39,6 +49,86 @@ public void test(String opHost, String redirectUrl, String userId, String userSe
         refreshToken(tokensResponse, client, site);
     }
 
+    @Parameters({"opHost", "redirectUrl", "userId", "userSecret"})
+    @Test
+    public void testWithInvalidAuthorizationCode(String opHost, String redirectUrl, String userId, String userSecret) throws Exception{
+
+        final DevelopersApi client = Tester.api();
+
+        final RegisterSiteResponse site = RegisterSiteTest.registerSite(client, opHost, redirectUrl);
+
+        final String state = CoreUtils.secureRandomString();
+
+        final String authorizationStr = Tester.getAuthorization(site);
+
+        final String code = codeRequest(client, site.getOxdId(), userId, userSecret, state, CoreUtils.secureRandomString(), authorizationStr);
+
+        final String invalidCode = code.concat("blah");
+
+        final GetTokensByCodeParams params = new GetTokensByCodeParams();
+        params.setOxdId(site.getOxdId());
+        params.setCode(invalidCode);
+        params.setState(state);
+
+        try {
+            client.getTokensByCode(authorizationStr, params);
+        } catch (ApiException ex) {
+            assertEquals(400, ex.getCode());
+        }
```

Comment:
so if exception is not thrown (we didn't get 400) then test will not fail here. It must fail if exception is not thrown too.